File: Server/src/main/java/lk/wnl/wijeya/entity/ProductMaterial.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "product_material")
public class ProductMaterial {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "quentity")
    private Integer quantity;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "material_id", nullable = false)
    private Material material;

    @Column(name = "line_cost", precision = 7, scale = 2)
    private BigDecimal lineCost;

}

File: Client/src/app/entity/ProductMaterial.ts
import {Material} from "./Material";

export class ProductMaterial{
  lineCost?: number;
  id?: number;
  name?: string;
  material?: Material;
  quantity?: number;


  constructor(id: number, material: Material,lineCost: number,  quantity: number) {
    this.lineCost = lineCost;
    this.id = id;
    this.material = material;
    this.quantity = quantity;
  }
}
